src/Microsoft.DotNet.Build.Tasks/PackageFiles/FrameworkTargeting.targets

```diff
@@ -64,7 +64,7 @@
 
   <Target Name="UpdateReferenceItems"
           DependsOnTargets="SetupDefaultReferences"
-          BeforeTargets="BeforeResolveReferences"
+          BeforeTargets="BeforeResolveReferences;ResolveAssemblyReferences"
```

Comment:
Why two here?  Usually having multiple entries for BeforeTargets is a bad idea because it means you'll run before one or the other, whichever comes first.

Reply:
The issue is that the new VS C# project system runs targets in a different order than they are run during a normal `Build`.

I don't want to break the existing/legacy .csprojs by moving this target around (thus why I left `BeforeTargets=BeforeResolveReferences`).

However, when doing design-time builds in the new project system, `BeforeResolveReferences` is running AFTER `ResolveAssemblyReferences`.  Thus `UpdateReferenceItems` is happening too late (after RAR), and so the References we are adding here aren't getting picked up.

I'll add a comment here to describe why we are doing it before both targets (in order to support the legacy format, normal builds, and the new C# project system).

Reply:
Should we create a AddReferencesDynamically target similar to AddProjectReferencesDynamically to be a placeholder for how to setup the references?

Reply:
I can if you want.  Do you think it will be valuable going forward?

Reply:
I suspect it will be valuable going forward as we will likely still do some similar default referencing.